src/Numerics.Tests/RootFindingTests/FindRootsTest.cs

```diff
@@ -207,8 +208,101 @@ public void QuadraticExpected(double c, double b, double a, double x1R, double x
 
             // Verify they really are roots
-            AssertComplexEqual(Complex.Zero, c + b*x1 + a*x1*x1, 1e-14);
-            AssertComplexEqual(Complex.Zero, c + b*x2 + a*x2*x2, 1e-14);
+            AssertComplexEqual(Complex.Zero, c + b * x1 + a * x1 * x1, 1e-14);
+            AssertComplexEqual(Complex.Zero, c + b * x2 + a * x2 * x2, 1e-14);
+        }
+
+        public static (Complex, Complex) FasterQuadratic(double c, double b, double a)
+        {
+            Complex x1, x2;
+            if (a == 0.0)
+            {
+                if (b == 0.0)
+                {
+                    x1 = new Complex(double.NaN, double.NaN);  // Complex.NaN;
+                    x2 = x1;
+                }
+                else
+                {
+                    x1 = new Complex(-c / b, 0.0);
+                    x2 = x1;
+                }
+            }
+            else
+            {
+                a = 1.0 / a;
+                b = -0.5 * b * a;
+                c = c * a;
+                double delta = b * b - c;
+                if (delta < 0.0)
+                {
+                    double sqrtDelta = Math.Sqrt(-delta);
+                    x1 = new Complex(b, sqrtDelta);
+                    x2 = new Complex(b, -sqrtDelta);
+                }
+                else
+                {
+                    double sqrtDelta = Math.Sqrt(delta);
+                    x1 = new Complex(b + sqrtDelta, 0.0);
+                    x2 = new Complex(b - sqrtDelta, 0.0);
+                }
+            }
+            return (x1, x2);
+        }
+
+        [TestCase(70_000_000)]
+        public void TestQuadraticSpeed(int N)
+        {
+            Complex x1, x2;
+            double sum = 0.0;
+            double a, b, c;
+            Stopwatch stopwatch = new Stopwatch();
+            for (int i = 11; i < N; i++)
+            {
+                a = -0.1 * i + 1.0;
+                b = 0.3464 * i + 5.0;
+                c = -0.3 * i + 7.0;
+                (x1, x2) = FindRoots.Quadratic(c, b, a);
+                sum += x1.Real + x2.Imaginary;
+            }
+            Console.WriteLine($"sum={sum}");
+            stopwatch.Stop();
+            Console.WriteLine($"FindRoots.Quadratic time: {stopwatch.ElapsedMilliseconds * 0.001:F3}s");
+
+            sum = 0.0;
+            stopwatch.Restart();
+            for (int i = 11; i < N; i++)
+            {
+                a = -0.1 * i + 1.0;
+                b = 0.3464 * i + 5.0;
+                c = -0.3 * i + 7.0;
+                (x1, x2) = FasterQuadratic(c, b, a);
+                sum += x1.Real + x2.Imaginary;
+            }
+            Console.WriteLine($"sum={sum}");
+            stopwatch.Stop();
+            Console.WriteLine($"FasterQuadratic time: {stopwatch.ElapsedMilliseconds * 0.001:F3}s");
+        }
+
+        [TestCase(7_000_000)]
+        public void TestFasterQuadraticCorrectness(int N)
```

Comment:
I don't think these tests bring some value to the quality control of the code base. Performance benchmarks are indeed significant, but the results are not tested. And the correctness test is checking the same results, since the faster quadratic implementation is the same as the production code